contracts/governance/TimelockController.sol

```diff
@@ -31,6 +32,43 @@ contract TimelockController is AccessControl, IERC721Receiver, IERC1155Receiver
     mapping(bytes32 => uint256) private _timestamps;
     uint256 private _minDelay;
 
+    enum OperationState {
+        Unset,
+        Pending,
+        Ready,
+        Done
+    }
+
+    /**
+     * @dev Mismatch between the parameters length for a operation call.
+     */
+    error TimelockInvalidOperationLength(uint256 targets, uint256 payloads, uint256 values);
+
+    /**
+     * @dev The schedule operation doesn't met the minimum delay.
+     */
+    error TimelockInsufficientDelay(uint256 delay, uint256 minDelay);
+
+    /**
+     * @dev The current state of an operation is not the required.
+     */
+    error TimelockIncorrectState(bytes32 operationId, OperationState expected);
```

Comment:
Same as in Governor, what do you think about "unexpected" or "unsupported" instead of "incorrect"?

Reply:
We decided to use `TimelockUnexpectedOperationState` according to the comments on Governor.